src/lib/spsp.js

```diff
@@ -68,77 +93,65 @@ const _quote = function * ({ plugin, spsp, sourceAmount, destinationAmount, ilp
+    sharedSecret: Buffer.from(payment.spsp.shared_secret, 'base64'),
+    destinationAmount: payment.destinationAmount,
+    destinationAccount: payment.destinationAccount,
+    data: payment.data // optional
   })
-}
 
-const quoteDestination = (plugin, receiver, amount, ilp) => {
-  return co(function * () {
-    const spsp = yield _querySPSP(receiver)
-    // quote by the destination amount, leaving source amount unspecified
-    const quote = yield _quote({ plugin, spsp, destinationAmount: amount, ilp })
-    return _createPayment(spsp, quote)
+  const fulfill = new Promise((resolve, reject) => {
+    function remove () {
+      plugin.removeListener('outgoing_fulfill', fulfill)
+      plugin.removeListener('outgoing_cancel', cancel)
+      plugin.removeListener('outgoing_reject', cancel)
+    }
+
+    function fulfill (transfer, fulfillment) {
+      if (transfer.id !== payment.id) return
+      remove()
+      resolve({ fulfillment })
+    }
+
+    function cancel (transfer) {
+      if (transfer.id !== payment.id) return
+      remove()
+      reject(new Error('transfer ' + payment.id + ' failed.'))
+    }
+
+    plugin.on('outgoing_fulfill', fulfill)
+    plugin.on('outgoing_cancel', cancel)
+    plugin.on('outgoing_reject', cancel)
   })
-}
 
-const sendPayment = (plugin, payment, ilp) => {
-  return co(function * () {
-    if (!plugin) throw new Error('missing plugin')
-    if (!payment) throw new Error('missing payment')
-    if (!payment.spsp) throw new Error('missing SPSP response in payment')
-    if (!payment.spsp.shared_secret) throw new Error('missing SPSP shared_secret')
-    if (!payment.destinationAmount) throw new Error('missing destinationAmount')
-    if (!payment.sourceAmount) throw new Error('missing sourceAmount')
-    if (!payment.destinationAccount) throw new Error('missing destinationAccount')
-    if (!payment.id) throw new Error('payment must have an id')
-
-    const sender = Sender.createSender(Object.assign({
-      client: (new IlpCore.Client(plugin))
-    }, ilp))
-
-    const request = sender.createRequest({
-      id: payment.id,
-      sharedSecret: payment.spsp.shared_secret,
-      destinationAmount: payment.destinationAmount,
-      destinationAccount: payment.spsp.destination_account,
-      data: payment.data
-    })
-
-    const fulfillment = yield sender.payRequest({
-      uuid: payment.id,
-      sourceAmount: String(payment.sourceAmount),
-      connectorAccount: payment.connectorAccount,
-      destinationAmount: String(payment.destinationAmount),
-      destinationAccount: request.address,
-      destinationMemo: {
-        data: request.data,
-        expires_at: request.expires_at
-      },
-      executionCondition: request.condition,
-      expiresAt: request.expires_at
-    })
-
-    return { fulfillment }
+  yield plugin.sendTransfer({
+    id: payment.id,
+    // TODO: should connectorAccount become destinationAccount when
+    // there is no connector?
```

Comment:
If the only reason that would happen is if it's a same-ledger transfer, then yes